scm-webapp/src/main/java/sonia/scm/metrics/AuthenticationMetrics.java

```diff
@@ -0,0 +1,137 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package sonia.scm.metrics;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+
+public class AuthenticationMetrics {
+
+  private AuthenticationMetrics() {
+  }
+
+  /**
+   * Creates counter to track amount of login attempts to SCM-Manager.
+   *
+   * @param registry meter registry
+   * @param type     type of login e.g.: api_key, bearer_token, etc.
+   * @return new {@link Counter}
+   */
+  public static Counter loginAttempts(MeterRegistry registry, String type) {
+    return Counter
+      .builder("scm.auth.login.attempts")
+      .description("The amount of login attempts to SCM-Manager")
+      .tags("type", type)
+      .register(registry);
+  }
+
+  /**
+   * Creates counter to track amount of failed logins to SCM-Manager.
+   *
+   * @param registry meter registry
+   * @param type     type of failed login, e.g.: UI/REST
+   * @return new {@link Counter}
+   */
+  public static Counter loginFailed(MeterRegistry registry, String type) {
+    return Counter
+      .builder("scm.auth.login.failed")
+      .tags("type", type)
+      .description("The amount of failed logins to SCM-Manager")
+      .register(registry);
+  }
+
+  /**
+   * Creates counter to track amount of logouts to SCM-Manager.
+   *
+   * @param registry meter registry
+   * @param type     type of logout, e.g.: UI/REST
+   * @return new {@link Counter}
+   */
+  public static Counter logout(MeterRegistry registry, String type) {
+    return Counter
+      .builder("scm.auth.logout")
+      .description("The amount of logouts from SCM-Manager")
+      .tags("type", type)
+      .register(registry);
+  }
+
+  /**
+   * Creates counter to track amount of token refreshes by SCM-Manager.
+   *
+   * @param registry meter registry
+   * @param type     type of refreshed token, e.g.: JWT
+   * @return new {@link Counter}
+   */
+  public static Counter tokenRefresh(MeterRegistry registry, String type) {
+    return Counter
+      .builder("scm.auth.token.refresh")
+      .description("The amount of authentication token refreshes")
+      .tags("type", type)
+      .register(registry);
+  }
+
+  /**
+   * Creates counter to track amount of successful accesses to SCM-Manager realms with token.
+   *
+   * @param registry meter registry
+   * @param realm    type of realm e.g.: {@link sonia.scm.security.BearerRealm}
+   * @param token    type of token e.g.: {@link org.apache.shiro.authc.UsernamePasswordToken},
+   * @return new {@link Counter}
+   */
+  public static Counter accessRealmSuccessful(MeterRegistry registry, String realm, String token) {
+    return Counter
+      .builder("scm.auth.access.successful")
```

Comment:
Please use a dedicated metric for realm success.

```suggestion
      .builder("scm.auth.realm.successful")
```